From d1df9352b217a0fdc19beef65acfbfeeadc5219d Mon Sep 17 00:00:00 2001
From: Jeremy Roman <jbroman@chromium.org>
Date: Thu, 2 Feb 2023 08:25:59 +0000
Subject: [PATCH] Make Document::node_count_ itself DCHECK-only.

Trivial followup from
  https://chromium-review.googlesource.com/c/chromium/src/+/4209074

Bug: 808503
Change-Id: Ia0db8371ad49bef2ee1ee66034721acf62310d95
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4214724
Reviewed-by: Camillo Bruni <cbruni@chromium.org>
Auto-Submit: Jeremy Roman <jbroman@chromium.org>
Commit-Queue: Camillo Bruni <cbruni@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1100304}
---
 third_party/blink/renderer/core/dom/document.cc | 1 -
 third_party/blink/renderer/core/dom/document.h  | 5 +++--
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/third_party/blink/renderer/core/dom/document.cc
+++ b/third_party/blink/renderer/core/dom/document.cc
@@ -805,7 +805,6 @@
           this,
           &Document::DidAssociateFormControlsTimerFired),
       parser_sync_policy_(kAllowDeferredParsing),
-      node_count_(0),
       // Use the source id from the document initializer if it is available.
       // Otherwise, generate a new source id to cover any cases that don't
       // receive a valid source id, this for example includes but is not limited
--- a/third_party/blink/renderer/core/dom/document.h
+++ b/third_party/blink/renderer/core/dom/document.h
@@ -1640,7 +1640,6 @@
     DCHECK_GT(node_count_, 0);
     node_count_--;
   }
-  int NodeCount() const { return node_count_; }
 #endif  // DCHECK_IS_ON()
 
   SnapCoordinator& GetSnapCoordinator();
@@ -2463,7 +2462,9 @@
 
   Member<IntersectionObserverController> intersection_observer_controller_;
 
-  int node_count_;
+#if DCHECK_IS_ON()
+  int node_count_ = 0;
+#endif
 
   Member<SnapCoordinator> snap_coordinator_;
 
